### PR TITLE
TimeUtils: migrate `dateTimeFormat()` method to `java.time` API

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/base/internal/TimeUtils.java [hunk 1]
```diff
@@ -124,21 +122,9 @@ public static Duration longest(Duration duration1, Duration duration2) {
...
-    }
-
-    /**
-     * Formats a given date+time value to an ISO 8601 string.
-     * @param dateTime value to format, unix time (ms)
-     */
-    public static String dateTimeFormat(long dateTime) {
-        DateFormat iso8601 = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
-        iso8601.setTimeZone(UTC);
-        return iso8601.format(dateTime);
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/base/internal/TimeUtils.java:142]: Oh, you're not messing around! The old methods are going away!

**Review by msgilligan — APPROVED** [review]
Looks great!

File: core/src/main/java/org/bitcoinj/base/internal/TimeUtils.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/base/internal/TimeUtils.java:142]: Yes, the API is internal, right?

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**